client/components/App.jsx

```diff
@@ -16,27 +16,41 @@ const App = () => {
   const [isLoading, setIsLoading] = useState(true);
   const [isError, setIsError] = useState({ error: false, msg: '' });
   const [rating, setRating] = useState(0);
+  const [productId, setProductId] = useState(INITIAL_PRODUCT_ID);
 
   useEffect(() => {
-    axios.get(`/api/products/${INITIAL_PRODUCT_ID}`)
+    setIsLoading(true);
+    axios.get(`/api/products/${productId}`)
       .then(({ data }) => {
         setCurrentProduct(data);
         getStarRatingAvg(data.id)
           .then((result) => setRating(result.avg));
         setIsLoading(false);
       })
       .catch((err) => {
+        setIsLoading(false);
         setIsError({ error: true, msg: err });
       });
-  }, []);
+  }, [productId]);
+
+  const handleCardClick = (id) => {
+    setProductId(id);
+  };
 
   return (
     <div className="app__container">
       <header className="app__header">Logo and Search Go Here</header>
-      {isLoading ? <p>Loading!</p> : <ProductOverview productId={currentProduct.id} />}
-      {isLoading ? <p>Loading!</p>
-        : <RelatedProducts rating={rating} currentProduct={currentProduct} />}
-      {currentProduct && <ReviewRating id={currentProduct.id} />}
+      {!isLoading ? (
```

Comment:
All three of our components were waiting on the load from the API, so I combined them to make it cleaner.

Reply:
Nice refactoring, I also handled the merge conflict with the axios call and the getStarRating component. getStarRating now passes both the average review and review count properly to components.